apps/web/components/CheckoutAddress/CheckoutAddress.vue

```diff
@@ -2,21 +2,29 @@
   <div data-testid="checkout-address" class="md:px-4 py-6">
     <div class="flex justify-between items-center">
       <h2 class="text-neutral-900 text-lg font-bold mb-4">{{ heading }}</h2>
-      <SfButton v-if="savedAddress" size="sm" variant="tertiary" @click="open">
+      <SfButton v-if="addresses.length > 0" size="sm" variant="tertiary" @click="edit">
         {{ $t('contactInfo.edit') }}
       </SfButton>
     </div>
 
-    <div v-if="savedAddress" class="mt-2 md:w-[520px]">
-      <p>{{ `${userAddressGetters.getFirstName(savedAddress)} ${userAddressGetters.getLastName(savedAddress)}` }}</p>
-      <p>{{ userAddressGetters.getPhone(savedAddress) }}</p>
-      <p>{{ userAddressGetters.getStreetName(savedAddress) }} {{ userAddressGetters.getStreetNumber(savedAddress) }}</p>
-      <p>{{ `${userAddressGetters.getProvince(savedAddress)} ${userAddressGetters.getPostCode(savedAddress)}` }}</p>
+    <div v-if="addresses.length > 0 && isAuth" class="mt-2 md:w-[520px]">
+      <SfSelect v-model="selectedAddress">
+        <option v-for="(address, index) in addresses" :key="index" :value="address.id?.toString()">
+          {{ `${userAddressGetters.getFirstName(address)} ${userAddressGetters.getLastName(address)}` }}
+        </option>
+      </SfSelect>
     </div>
 
-    <div v-else class="w-full md:max-w-[520px]">
-      <p>{{ description }}</p>
-      <SfButton class="mt-4 w-full md:w-auto" variant="secondary" @click="open">
+    <div v-if="addresses.length === 1 && !isAuth" class="mt-2 md:w-[520px]">
+      <p>{{ `${userAddressGetters.getFirstName(addresses[0])} ${userAddressGetters.getLastName(addresses[0])}` }}</p>
```

Comment:
**suggestion (non-blocking)**: Extract `addresses[0]` to something like `currentAddress` to improve readability.